tests/Commands/ToggleDone.test.ts

```diff
@@ -227,5 +227,21 @@ describe('ToggleDone', () => {
         });
     });
 
+    it('should proceed through the statuses until a TODO status is reached when a task is completed', () => {
+        // Arrange
```

Comment:
To confirm that there was indeed a loop, I set up this status set:

![image](https://github.com/obsidian-tasks-group/obsidian-tasks/assets/4840096/85aa65c1-59e6-458a-a215-55bb4995a9a9)

And then toggled this task:

```
- [1] #task Do stuff 🔁 every day 📅 2023-12-19
```

Obsidian did indeed become unresponsive when I toggled the task.

So that would be good data to use in a new test.